carla/recourse_methods/catalog/face/library/face_method.py

```diff
@@ -0,0 +1,229 @@
+# import helpers
+import numpy as np
+
+# import Dijkstra's shortest path algorithm
+from scipy.sparse import csgraph, csr_matrix
+
+# import graph building methods
+from sklearn.neighbors import kneighbors_graph, radius_neighbors_graph
+
+
+def graph_search(
```

Comment:
This is one large function with 200 LOC, can we split this up in subfunctions? It is very hard to grasp as a reader what actually happens here